src/Handlers/PaymentMethodHandler.php

```diff
@@ -55,4 +128,155 @@ public function finalize(
+            $stateData['browserInfo']['screenHeight'],
+            $stateData['browserInfo']['colorDepth'],
+            $stateData['browserInfo']['timeZoneOffset'],
+            $stateData['browserInfo']['language'],
+            $stateData['browserInfo']['javaEnabled'],
+            $request
+        );
+        $request = $this->addressBuilder->buildDeliveryAddress(
+            $shippingStreetAddress['street'],
+            $shippingStreetAddress['houseNumber'],
+            $salesChannelContext->getShippingLocation()->getAddress()->getZipcode(),
+            $salesChannelContext->getShippingLocation()->getAddress()->getCity(),
+            $shippingState,
+            $salesChannelContext->getShippingLocation()->getAddress()->getCountry()->getIso(),
+            $request
+        );
+        $request = $this->addressBuilder->buildBillingAddress(
+            $billingStreetAddress['street'],
+            $billingStreetAddress['houseNumber'],
+            $salesChannelContext->getCustomer()->getActiveBillingAddress()->getZipcode(),
+            $salesChannelContext->getCustomer()->getActiveBillingAddress()->getCity(),
+            $billingState,
+            $salesChannelContext->getCustomer()->getActiveBillingAddress()->getCountry()->getIso(),
+            $request
+        );
+        $request = $this->paymentBuilder->buildPaymentData(
+            $salesChannelContext->getCurrency()->getIsoCode(),
+            $this->currency->sanitize(
+                $transaction->getOrder()->getPrice()->getTotalPrice(),
+                $salesChannelContext->getCurrency()->getIsoCode()
+            ),
+            $transaction->getOrder()->getOrderNumber(),
+            $this->configurationService->getMerchantAccount(),
+            $transaction->getReturnUrl(),
+            $request
+        );
+        $request = $this->customerBuilder->buildCustomerData(
+            false,
+            $salesChannelContext->getCustomer()->getEmail(),
+            $salesChannelContext->getShippingLocation()->getAddress()->getPhoneNumber(),
+            '',
+            $customerBirthday,
+            $salesChannelContext->getCustomer()->getFirstName(),
+            $salesChannelContext->getCustomer()->getLastName(),
+            $salesChannelContext->getShippingLocation()->getAddress()->getCountry()->getIso(),
+            'en-GB', //TODO replace with function in generic component branch
+            $salesChannelContext->getCustomer()->getRemoteAddress(),
+            $salesChannelContext->getCustomer()->getId(),
+            $request
+        );
+        $request = $this->paymentBuilder->buildCardData(
+            $stateData['paymentMethod']['encryptedCardNumber'],
+            $stateData['paymentMethod']['encryptedExpiryMonth'],
+            $stateData['paymentMethod']['encryptedExpiryYear'],
+            $stateData['paymentMethod']['holderName'],
+            'http://192.168.33.10', //TODO replace with util function in generic component branch
```

Comment:
try skipping it - otherwise a builder needs to be used for this
But best you send it from the frontend inside the request as an additional info

Reply:
Now the origin comes from the frontend JS